include/util/extract-conditionals.hpp

```diff
@@ -0,0 +1,386 @@
+#include "util/conditional_restrictions.hpp"
```

Comment:
Somehow I ended up with an old copy of the same file in `include/conditionals/` that did have an include guard. I've deleted that one and added the guard to this file.